### ISSUE TITLE
Feature request : Add an option to disable Admin portal and SSL

### ISSUE THREAD

**Issue by christopheblin** [mrak/stubby4net#9]
At the moment, we are using stubby in our automated integration tests.

Doing this, we have no use for the admin portal and for TLS.

Normally, this would not cause any problem to start them anyway but sometimes we have this kind of errors :

Initialization method Example.Tests.SitesControllerTest.InitHttpStubAndWebApi threw exception. System.Net.HttpListenerException: System.Net.HttpListenerException: L'‚coute du pr‚fixe 'http://localhost:8889/' a ‚chou‚ parce que le pr‚fixe est en conflit avec un enregistrement existant sur l'ordinateur..
Stack Trace:
    … System.Net.HttpListener.AddAllPrefixes()
   … System.Net.HttpListener.Start()
   … stubby.Portals.Admin.Start(String location, UInt32 port)
   … stubby.Stubby.StartPortals()
   … stubby.Stubby.Start()

This is in french but in plain english it says that it can not start listening on 8889 because there is already another listener.

The fact is that we are correctly closing stubby after our tests but it seems that the closing of the port is taking some milliseconds (and so when the next test is started and reopen the same port, there is the problem).

This problem only occurs when we launch the tests via the vstest.console.exe, not via VisualStudio...

By only starting the necessary listener, we hope that this will not occur anymore.

To my understanding, this is not hard to do (i.e modifying Stubby.cs), but I have the concern that you may not want to implement that "feature" in regards to another matter that is out of my knowledge.
I can do a pull request if you agree to this feature.

**Comment by mrak** [comment]
Thanks for reporting!

Your use case makes a lot of sense to me. I'll be happy to look at a PR that makes the admin portal and TLS optional.

**Comment by christopheblin** [comment]
Do you prefer that
1. IArguments.Admin becomes a uint? (same for Tls) and when it is null it means the port is not opened
2. adding a boolean IArguments.EnableAdmin with default value true (same for Tls)
3. adding a boolean IArguments.DisableAdmin with default value false (same for Tls)

I would say that 3 is the best option, what do you think ?

**Comment by mrak** [comment]
Probably option 3 to avoid explicit use of `null`. Ideally, we'd have some sort of built-in `Option<unit>` type in C#.

**Comment by christopheblin** [comment]
PR merged => can be closed !
